work_items/cbb_policy_as_code.md

```diff
@@ -0,0 +1,49 @@
+# Crypto-systems Building Blocks: Policy-as-Code Language
+
+This is the eigth proposed work item in a series of work items under the
+category of "crypto-systems building blocks".
+
+This work item is to propose an IETF standard for an application independent,
+non-Turing-complete language and virtual machine definition for encoding and
+executing cryptographic policies as code (i.e. smart contracts). Fully
+decentralized crypto-systems require a method for defining policies that allow
+for decentralized and independent verification and enforcement. This is similar
+to the way Bitcoin users define the policy for validating Bitcoin transactions
+such that Bitcoin miners can do the validation independently and without any
+coordination with other miners. This is a generalization of that concept to
+application neutral, authentic data applications that require maximum
+decentralization and scalability.
+
+## Work Item Owners
+- Dave Huseby (@dhuseby, dave@cryptid.tech, CryptID Technologies, Inc)
+- Wayne Chang (@wyc, Spruce Systems, Inc)
+
+## External Partners
+- IETF, new WG for crypto-systems building blocks?
+
+## Outcome
+- Establish a standard language definition for a non-Turing-complete
+  programming language designed for defining policy-as-code.
+- Establish a standard virtual machine definition that formalizes external
+  data sources that may be referenced by the language at run-time.
+
+## Deliverables
+- An IETF formatted draft standard, ready for proposal.
+- A white paper describing the design decisions (e.g. non-Turing-complete) as
+  well as the applications of the language in provenance logs and other crypto-
+  systems (e.g. emailed Git patches).
+- One or more reference implementations.
+- The expected timeline is 1-3 months to completion.
+- The audience is any developer that works with crypto-systems and requires a
+  method for defining late-binding policies as code for decentralized and
+  independent verification of cryptographically secure data.
+
+### Relevant Related Standards
+- [Crypto Construct Language][0]
+- [Bitcoin Script][1]
+
+## Meetings
+- Implementor updates are presented at the regular Applied Crypto WG meetings.
+
+[0]: https://github.com/dhuseby/cclang
+[1]: https://en.bitcoin.it/wiki/Script
```

Comment:
```suggestion
[1]: https://en.bitcoin.it/wiki/Script
[2]: https://github.com/w3c-ccg/verifiable-conditions
```

Reply:
I'll add the reference but what we're proposing here is not functional cryptographic constructs which is what CCLang was originally designed for. I only include a reference to CCLang because it is implemented on a generic stack machine that could form the foundation of this new non-Turing-complete policy-as-code language.

I really want to avoid allowing any W3C designs to influence this. We're targeting the IETF. This is not a web standard and I'm reluctant to do anything linked data related. i think of this as being more general than that and making changes for compatibility with W3C/webby stuff IMO would compromise the goals of this work item. That's just my 2p. Just wanted to share my expectations.